app/src/main/java/ca/bc/gov/bchealth/ui/healthpass/HealthPassFragment.kt

```diff
@@ -248,7 +302,33 @@ class HealthPassFragment : Fragment(R.layout.fragment_my_cards) {
         val btnAddCardOptions: MaterialButton =
             sceneNoCardPlaceHolder.sceneRoot.findViewById(R.id.btn_add_card)
         btnAddCardOptions.setOnClickListener {
-            findNavController().navigate(R.id.addCardOptionFragment)
+            sharedViewModel.destinationId = R.id.addCardOptionFragment
+            checkLogin()
+        }
+    }
+
+    private fun checkLogin() {
+        bcscAuthViewModel.checkLogin()
+
+        viewLifecycleOwner.lifecycleScope.launch {
```

Comment:
check if we can merge this call with the existing flow observer.

Reply:
Could not merge as this observer is only required when user clicks on add card option.